pkg/reconciler/reconciler.go

```diff
@@ -846,7 +857,50 @@ func (r *Reconciler) doUninstall(actionClient helmclient.ActionInterface, u *upd
 		if log.V(4).Enabled() {
 			fmt.Println(diff.Generate(resp.Release.Manifest, ""))
 		}
+
+		if !wait {
+			updater.EnsureCondition(conditions.Deployed(corev1.ConditionFalse, conditions.ReasonUninstallSuccessful, ""))
+		}
+	}
+
+	if wait {
+		updater.EnsureCondition(conditions.Deployed(corev1.ConditionFalse, conditions.ReasonUninstallSuccessful, "Waiting until all resources are deleted."))
+	}
+
+	if wait && resp.Release != nil && resp.Release.Manifest != "" {
+		log.Info("Uninstall wait")
+		config, err := config.GetConfig()
+		if err != nil {
+			log.Error(err, "Failed to get REST config")
+			updater.EnsureCondition(conditions.ReleaseFailed(corev1.ConditionTrue, conditions.ReasonUninstallError, err.Error()))
+			return err
+		}
+		restMapper, err := apiutil.NewDynamicRESTMapper(config)
+		if err != nil {
+			log.Error(err, "Failed to get RESTMapper")
+			updater.EnsureCondition(conditions.ReleaseFailed(corev1.ConditionTrue, conditions.ReasonUninstallError, err.Error()))
+			return err
+		}
+		isAllResourcesDeleted := false
+		for !isAllResourcesDeleted {
+			isAllResourcesDeleted, err = CleanupRelease(helmclient.NewActionConfigGetter(config, restMapper, log), obj, resp.Release.Manifest, log)
+			if err != nil {
+				log.Error(err, "Failed to cleanup release")
+				updater.EnsureCondition(conditions.ReleaseFailed(corev1.ConditionTrue, conditions.ReasonUninstallError, err.Error()))
+				return err
+			}
+			if !isAllResourcesDeleted {
+				log.Info("Waiting until all resources are deleted")
+
+				// wait for a little bit of time before looping again
+				time.Sleep(time.Second * 5)
```

Comment:
We probably shouldn't block like this in the reconcile function. Can we just requeue the reconcile request and check again the next time around?